src/Data/Vec/Base.agda

```diff
@@ -284,3 +287,10 @@ init .(ys ∷ʳ y) | (ys , y , refl) = ys
 last : ∀ {n} → Vec A (1 + n) → A
 last xs         with initLast xs
 last .(ys ∷ʳ y) | (ys , y , refl) = y
+
+------------------------------------------------------------------------
+-- Other operations
+
+transpose : ∀ {m n} → Vec (Vec A n) m → Vec (Vec A m) n
```

Comment:
Can't help feeling this should be in a `Matrix` class... Don't have any good ideas about what to do with it at the moment though! Just thought I'd put it out there.